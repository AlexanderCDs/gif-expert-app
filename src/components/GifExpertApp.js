/**
 * @author: alexandercd
 * @date: 22/septiembre/2021
 * @description: 
**/
import React, {useState} from 'react';
import AddCategory from './AddCategory';
import GifGrid from './GifGrid';

const GifExpertApp = ({defaultData = []}) => {
    const [categories, setCategories] = useState(defaultData); 
    return (
        <>
            <div className="center">
                <h2>Gif Expert App</h2>
                <AddCategory 
                    placeholder="Ingrese un texto"
                    setCategories={setCategories}
                />
            </div> 
            <ol>
                {categories.map((category, index) => {
                    return (
                        <GifGrid 
                            key={category} 
                            category={category} 
                            limit={20}
                        />
                    )
                })}
            </ol>
        </>
    )
} 

export default GifExpertApp;