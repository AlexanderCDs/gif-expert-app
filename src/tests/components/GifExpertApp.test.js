/**
 * @author: alexandercd
 * @date: 27/septiembre/2021
 * @description: 
**/
import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../../components/GifExpertApp';  

describe('Test <GifGrid />', () => { 
    const defaultData = [
        'One Punch',
        'Dragon Ball'
    ]

    test('should show <GifExpertApp /> success', () => { 
        const wrapper = shallow( <GifExpertApp />);
        expect( wrapper ).toMatchSnapshot();
    }); 

    test('should show list categories', () => { 
        const wrapper = shallow( <GifExpertApp defaultData={defaultData} />);
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( defaultData.length );
    }); 

})
