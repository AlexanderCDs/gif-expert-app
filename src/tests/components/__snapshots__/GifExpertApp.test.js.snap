// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGrid /> should show <GifExpertApp /> success 1`] = `
<Fragment>
  <div
    className="center"
  >
    <h2>
      Gif Expert App
    </h2>
    <AddCategory
      placeholder="Ingrese un texto"
      setCategories={[Function]}
    />
  </div>
  <ol />
</Fragment>
`;

exports[`Test <GifGrid /> should show list categories 1`] = `
<Fragment>
  <div
    className="center"
  >
    <h2>
      Gif Expert App
    </h2>
    <AddCategory
      placeholder="Ingrese un texto"
      setCategories={[Function]}
    />
  </div>
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
